src/fastfield/serializer/gcd.rs

```diff
@@ -0,0 +1,176 @@
+use fastdivide::DividerU64;
+use gcd::Gcd;
+
+pub const GCD_DEFAULT: u64 = 1;
+
+pub fn find_gcd(mut numbers: impl Iterator<Item = u64>, base: u64) -> Option<u64> {
+    let mut num1 = 0;
+    let mut num2 = 0;
+    loop {
+        let num = numbers.next()? - base;
```

Comment:
what is this loop doing? is it picking the two first non null integers?

Reply:
yep